### PR TITLE
Feature tabular meta

### PR DESCRIPTION
**Pull request by bewakes**

### CODE REVIEW

**Comment by bewakes** [comment]
Depends:
https://github.com/the-deep/server/pull/216

File: src/components/general/TriggerAndPoll/index.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ const defaultIsValid = r => r.status === 'success';
 
 const requests = {
     initialRequest: {
-        // schemaName: ({ props }) => props.initialSchemaName || props.schemaName,
+        schemaName: ({ props }) => props.initialSchemaName || props.schemaName,
```
Comment (tnagorra) [inline src/components/general/TriggerAndPoll/index.js]: Current rest request doesn't support dynamic schema Name

File: src/views/LeadAdd/LeadFormItem/LeadTabular/index.js [hunk 2]
```diff
@@ -35,22 +35,26 @@ export default class LeadTabular extends React.PureComponent {
     state = {
         bookId: undefined,
         fileType: undefined,
-        currentPage: 0,
+        metaInfo: undefined,
     }
 
     handleBookIdChange = (bookId, fileType, callback) => {
```
Comment (tnagorra) [inline src/views/LeadAdd/LeadFormItem/LeadTabular/index.js]: Looks like this is unused

File: src/views/LeadAdd/LeadFormItem/LeadTabular/AttributesPage/index.js [hunk 3]
```diff
@@ -28,49 +27,44 @@ import styles from './styles.scss';
...
+        onFailure: ({ error: { faramErrors }, params: { handleFaramError } }) => {
+            handleFaramError(faramErrors);
         },
         onFatal: ({ params: { handleFaramError } }) => {
             handleFaramError({
                 $internal: ['SERVER ERROR'],
             });
         },
     },
+
```
Comment (tnagorra) [inline src/views/LeadAdd/LeadFormItem/LeadTabular/AttributesPage/index.js]: Looks like saveBookRequest is also unused here

File: src/views/LeadAdd/LeadFormItem/LeadTabular/AttributesPage/index.js [hunk 4]
```diff
@@ -103,10 +97,13 @@ export default class AttributesPage extends React.PureComponent {
...
             faramErrors: {},
         };
+    }
 
-        this.props.createMetadataRequest.setDefaultParams({
-            initFaram: this.initFaram,
-            handleFaramError: this.handleFaramValidationFailure,
+    componentDidMount() {
+        const { metaInfo, defaultFileType } = this.props;
+        this.initFaram({
```
Comment (tnagorra) [inline src/views/LeadAdd/LeadFormItem/LeadTabular/AttributesPage/index.js]: This should be done in constructor

**Pull request merged**